src/s1reader/s1_orbit.py

```diff
@@ -1,12 +1,171 @@
+import cgi
 import datetime
 import glob
 import os
+import requests
 import warnings
 
+from xml.etree import ElementTree
 
 # date format used in file names
 FMT = "%Y%m%dT%H%M%S"
 
+# Scihub guest credential
+scihub_user = 'gnssguest'
+scihub_password = 'gnssguest'
+
+def check_internet_connection():
+    '''
+    Check connection availability
+    '''
+    url = "http://google.com"
+    try:
+        requests.get(url, timeout=10)
+    except (requests.ConnectionError, requests.Timeout) as exception:
+        raise ConnectionError(f'Unable to reach {url}: {exception}')
+
+
+def parse_safe_filename(safe_filename):
+    '''
+    Extract info from S1-A/B SAFE filename
+    SAFE filename structure: S1A_IW_SLC__1SDV_20150224T114043_20150224T114111_004764_005E86_AD02.SAFE
+    Parameters:
+    -----------
+    safe_filename: string
+       Path to S1-A/B SAFE file
+    Returns:
+    -------
+    List of [sensor_id, mode_id, start_datetime,
+                end_datetime, abs_orbit_num]
+       sensor_id: sensor identifier (S1A or S1B)
+       mode_id: mode/beam (e.g. IW)
+       start_datetime: acquisition start datetime
+       stop_datetime: acquisition stop datetime
+       abs_orbit_num: absolute orbit number
+    '''
+
+    safe_name = os.path.basename(safe_filename)
+    sensor_id = safe_name[2]
+    sensor_mode = safe_name[4:10]
```

Comment:
After I wrote out that example doctest, I saw that the `sensor_mode` actually doesn't match just "IW". right now it's returning "IW_SLC". 
Do we want both of those? Or should we change it to `safe_name[4:6]`?

Reply:
Good catch. We do not need the product level :)